spec/lib/safe_commit_spec.rb

```diff
@@ -0,0 +1,151 @@
+      options[:safety] = false
+      allow(ModiFile.instance).to receive(:expected_test_files).and_return(test_files_array)
+      expect(test_files(options)).to eq(test_files_array)
+    end
+  end
+
+  describe "#branch" do
+    it "returns the current branch name" do
+      branch_name = "main"
+      allow(self).to receive(:`).with("git branch --show-current").and_return("#{branch_name}\n")
+      expect(branch).to eq(branch_name)
+    end
+  end
+
+  describe "#trunk_branch" do
+    it "sets the protected branch for ModiFile instance" do
+      protected_branch = "main"
+      trunk_branch(protected_branch)
+      expect(ModiFile.instance.protected_branch).to eq(protected_branch)
+    end
+  end
+
+  describe "#be_trunk" do
+    it "returns the protected branch from ModiFile instance as a string" do
+      protected_branch = "main"
+      allow(ModiFile.instance).to receive(:protected_branch).and_return(protected_branch)
+      expect(be_trunk).to eq(protected_branch)
+    end
+  end
+
+  describe "#linting" do
+    it "returns the rubocop linting output" do
+      rubocop_output = "no offenses detected"
+      allow(Open3).to receive(:capture2).and_return([rubocop_output])
+      expect(linting).to eq(rubocop_output)
+    end
+  end
+
+  describe "#be_acceptable_enough" do
+    it "returns a string with the threshold of offenses" do
+      threshold = 5
+      expect(be_acceptable_enough(threshold)).to eq("below the threshold of #{threshold} offenses")
+    end
+  end
+
+  describe "#no_presence_of" do
+    let(:pattern) { "binding.pry" }
+    let(:message) { "There should be no bindings" }
+
+    context "when pattern is found in the diff" do
+      it "outputs the message and calls Assertion.instance.error" do
+        allow(Open3).to receive(:capture2).and_return(["1:#{pattern}\n"])
+        expect(Assertion.instance).to receive(:error).with(message)
+        no_presence_of(pattern, message)
+      end
+    end
+
+    context "when pattern is not found in the diff" do
+      it "does not call Assertion.instance.error" do
+        allow(Open3).to receive(:capture2).and_return([""])
+        expect(Assertion.instance).not_to receive(:error)
+        no_presence_of(pattern, message)
+      end
+    end
+  end
+
+  describe "#suggest_refactors" do
+    let(:modified_file) { "some_file.rb" }
+    let(:file_content) { "def some_method; end" }
+    let(:openai_api_key) { "test_key" }
+    let(:response) { double("response", dig: "suggestion") }
+
+    before do
+      allow(ENV).to receive(:fetch).with("SAFE_COMMIT_OPENAI_API_KEY").and_return(openai_api_key)
+      allow(File).to receive(:read).with(modified_file).and_return(file_content)
+      allow(self).to receive(:modified_files).and_return([modified_file])
+      allow(Interaction).to receive(:confirm).and_return(true)
+      allow(OpenAI::Client).to receive(:new).and_return(double("client", chat: response))
```

Comment:
Usually the Rspec cops complain about this (`RSpec/VerifiedDoubles: Prefer using verifying doubles over normal doubles.`). Is rubocop rspec set up correctly?